version: "3.8"

services:
  api_async:
    build: ../checkpoint/async/
    ports:
      - "8000:8000"
    container_name: api_async
    volumes:
      - ./logs:/app/logs

  logging:
    build: ./logging
    container_name: logging
    command: bash -c "cd app/ && python3 main.py" 
    environment:
      PYTHONPATH: .
      DATABASE_URL: "postgresql+asyncpg://username:password@postgres:5432/postgres"
    volumes:
      - ./logs:/app/app/logs
    ports:
      - '8003:8003'
    restart: on-failure


  filebeat:
    build: ./filebeat/
    container_name: filebeat
    volumes:
      - ./logs:/src/logs
    depends_on:
      - api_async
      - elasticsearch
      - kibana
    network_mode: "host"

  elasticsearch:
    build: ./elasticsearch/
    container_name: elasticsearch
    ports:
      - "9200:9200"
      - "9300:9300"

  kibana:
    build: ./kibana/
    container_name: kibana
    ports:
      - "5601:5601"
    depends_on:
      - elasticsearch
    # volumes:
    #     elasticsearch_data:
    #         driver: local

  gateway:
    build: ../Semana4/gateway
    restart: on-failure
    ports:
      - "8001:80"